zerver/lib/users.py

```diff
@@ -24,6 +26,20 @@ def check_short_name(short_name_raw: Text) -> Text:
         raise JsonableError(_("Bad name or username"))
     return short_name
 
+def check_valid_bot_config(service_name: str, config_data: Dict[str, str]) -> None:
+    try:
+        from zerver.lib.bot_lib import get_bot_handler
+        bot_handler = get_bot_handler(service_name)
+        if hasattr(bot_handler, 'validate_config'):
+            bot_handler.validate_config(config_data)
+    except ConfigValidationError:
+        # The exception provides a specific error message, but that
+        # message is not tagged translatable, because it is
+        # triggered in the external zulip_bots package.
+        # TODO: Think of some clever way to provide a more specific
+        # error message.
+        raise JsonableError(_("Incorrect configuration data!"))
```

Comment:
I think "Invalid configuration data" might be better text here.  But perhaps even better would be to pass something through from the bot, so it can tell them specifically what's wrong (e.g. "Invalid Giphy API  key")?